rusoto/core/src/request.rs

```diff
@@ -377,6 +377,7 @@ pub struct HttpConfig {
     read_buf_size: Option<usize>,
 }
 
+#[allow(clippy::new_without_default)]
```

Comment:
Should we address this? I put this in so `clippy` would look at the S3 crate - it was hung up on `rusoto_core` and wouldn't look at anything else until this was addressed.

Reply:
Seems like a simple enough fix to add a default impl that just calls `Self::new()`, and a small ergonomics win.

Reply:
Done!